### DISCUSSION TITLE
get version of each tool used in ATLAS

### DISCUSSION THREAD

**Discussion by Mechah** [metagenome-atlas/atlas#523] [Ideas]
Hey, is it possible to get respective versions of each tool from the ATLAS pipeline? Where can I find them and if not would it make sense to integrate this feature to improve replicability of a workflow?

**Comment by jmtsuji** [comment]
@Mechah One option is to use [snakemake reports](https://snakemake.readthedocs.io/en/stable/snakefiles/reporting.html) to summarize tool versions. Basically, just add the `--report report.html` argument at the end of the ATLAS run command, and an HTML file will be produced summarizing several run stats, including the workflow order (as discussed [here](https://github.com/metagenome-atlas/atlas/discussions/514#discussioncomment-2765913)) software versions.

Example:
```bash
atlas run -w . -c config.yaml -j 40 qc --report report.html
```

## Screenshots from my ATLAS report
These were made with ATLAS 2.8.2.

Workflow diagram:
<img width="1419" alt="atlas-reports-workflow" src="https://user-images.githubusercontent.com/18713012/170004861-1a2cb08f-78c7-4ca4-8843-2e008f2aacba.png">

You can click on a step in the workflow diagram and see versions:
<img width="1417" alt="atlas-reports-versions" src="https://user-images.githubusercontent.com/18713012/170004872-8110e204-42a5-4375-afc0-76a05238f850.png">

## Limitation
As one limitation, I think the versions listed in the snakemake report are the same as the versions specified in the YAML environment files for ATLAS (link [here](https://github.com/metagenome-atlas/atlas/tree/master/atlas/workflow/envs)). Some of these versions are not hard-coded but represent a range of possible tool versions. For example, for `bbmap` in the above screenshot, anything in version 37 is allowed. In reality, version 37.99 was used in my specific conda environment. Similarly, Python 3.8.12 was used in my env, although the snakemake report (and YAML file) only specify to version 3.8 (and leave the `.12` part) flexible.

Here are three possible workarounds for this limitation:
- You can manually activate the conda env made for a specific ATLAS step on your server and run `conda list` to get the exact version number. This is a bit of a pain but can work if you only have a few envs you are really interested in.
- All key tools in the ATLAS YAML environment files could have their exact versions hard coded... but this could make ATLAS break more easily over time (what do you think @SilasK ?)
- The Snakemake rules of ATLAS could manually export the exact conda versions, but I think this could get clunky -- see discussion [here](https://stackoverflow.com/a/64066220).

For the moment, though, I hope the ATLAS reports feature is helpful!

**Reply by Mechah** [reply]
Oh cool! I was not aware of the report function at all. Looks really helpful. Thanks for now. I do not think that versions need to be hard coded and I'm satisfied to look for specific versions with conda list (if this works - need to test it first).
What about adding this information to the docs? Or did I just overlook it?

**Reply by jmtsuji** [reply]
@Mechah Glad it helps! I don't think the `--report` is mentioned in the docs yet. (Support for report generation was just recently added to ATLAS, I think.) It could make sense to add it to the [useful-command-line-options](https://github.com/metagenome-atlas/atlas/blob/v2.9.1/docs/usage/getting_started.rst#useful-command-line-options) section. I'll try to make a PR for this sometime today.

**Reply by SilasK** [reply]
PR approved. Thank you @jmtsuji

**Reply by s-junguy** [reply]
Hi, does the report function also show the runtime? I generated a report but the plot with the runtime only shows zero. Is there any other way to get the runtime?
